build.sbt

```diff
@@ -2122,6 +2122,9 @@ lazy val `std-base` = project
   .settings(
     frgaalJavaCompilerSetting,
     autoScalaLibrary := false,
+    Compile / compile / compileInputs := (Compile / compile / compileInputs)
+      .dependsOn(SPIHelpers.ensureSPIConsistency)
```

Comment:
Could we do this for all projects that have `Compile / compile /compileInputs`?

Reply:
As noted in the docs - I'm not sure if this will work well with mixed-Scala projects - since as you noticed - I check for existence of a `.java` file corresponding to `.class`.

In case of mixed projects - that file may not exist if the service is written in Scala.

---

If we can guarantee we won't have such a case, I guess we could try to set it as a global setting, if I recall correctly it should be possible.

Reply:
I'd like to merge the current PR, because it solves the most immediate issue and got through CI.

But if we decide in the discussion that we want to amend how we 'register' this 'hook', I'm happy to do a followup PR updating that.